internal/jwt_utils.go

```diff
@@ -0,0 +1,90 @@
+package internal
+
+/*
+	Project Ricotta: Bechamel API
+
+	This file contains JWT (JSON Web Token) utility functions for the Bechamel API to use internally.
+*/
+
+import (
+	"errors"
+	"fmt"
+	"project-ricotta/bechamel-api/model"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+
+// TODO: get this from environment
+var jwtSigningKey = []byte("GetThisFromENV")
+
+// Number of seconds generated JWT tokens are valid for before expiration
+const JWT_TTL = 600
+
+/* Generates a JWT token for a given userName */
+func GenerateJWT(userName string) (string, error) {
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"userName": userName,
+		"nbf":      time.Now().Unix(),
+		"iat":      time.Now().Unix(),
+		"eat":      time.Now().Add(time.Second * time.Duration(JWT_TTL)).Unix(),
+	})
+	return token.SignedString(jwtSigningKey)
+}
+
+/*
+Validates a supplied JWT token and returns the userName for the user
+represented by the token if the token is valid and not expired.
+*/
+func verifyJWT(jwtTokenString string) (string, error) {
+	if jwtTokenString == "" {
+		return "", errors.New("jwt token to verify must be non-empty")
+	}
+
+	token, err := jwt.Parse(jwtTokenString, func(token *jwt.Token) (interface{}, error) {
+		return jwtSigningKey, nil
+	})
+	if token == nil || err != nil {
+		// TODO: propogate parsing error from JWT up in a more useful form?
+		return "", errors.New("could not parse supplied JWT token")
```

Comment:
I'd usually use `fmt.Errorf("could not parse supplied JWT: %w", err) here: we want to return an error type, properly wrapped, rather than printing to stdout